server/src/main/java/org/elasticsearch/snapshots/SnapshotShardsService.java

```diff
@@ -232,7 +232,7 @@ private void startNewSnapshots(SnapshotsInProgress snapshotsInProgress) {
                     if (snapshotStatus == null) {
                         // due to CS batching we might have missed the INIT state and straight went into ABORTED
                         // notify master that abort has completed by moving to FAILED
-                        if (shard.value.state() == ShardState.ABORTED) {
+                        if (shard.value.state() == ShardState.ABORTED && localNodeId.equals(shard.value.nodeId())) {
```

Comment:
This fixes a tricky bug that might have been trouble for aborts with pre-7.6 repository metadata on S3 and generally wastes a lot of shard status updates. I'll fix this in a separate PR right away, just adding a node here to not forget to handle that first.

Reply:
https://github.com/elastic/elasticsearch/pull/58214